three/three.h

```diff
@@ -5,27 +5,66 @@
+        = { PyModuleDef_HEAD_INIT, datadog_agent_six_##moduleName, NULL, -1, NULL, NULL, NULL, NULL, NULL };           \
+    PyMODINIT_FUNC PyInit_##moduleName(void) {                                                                         \
+        PyObject *m = PyModule_Create(&def_##moduleName);                                                              \
+        std::lock_guard<std::mutex> lock(Three::ModuleConstantsMtx);                                                   \
+        PyModuleConstants::iterator it = Three::ModuleConstants.find(moduleID);                                        \
+        if (it != Three::ModuleConstants.end()) {                                                                      \
+            std::vector<PyModuleConst>::iterator cit;                                                                  \
+            for (cit = it->second.begin(); cit != it->second.end(); ++cit) {                                           \
+                PyModule_AddIntConstant(m, cit->first.c_str(), cit->second);                                           \
+            }                                                                                                          \
+        }                                                                                                              \
+        return m;                                                                                                      \
+    }
+
+// APPEND_TO_PYTHON_INITTAB set the module methods and call
+// PyImport_AppendInittab with it, allowing Python to import it
+#define APPEND_TO_PYTHON_INITTAB(moduleID, moduleName)                                                                 \
+    {                                                                                                                  \
+        if (this->_modules[moduleID].size() > 0) {                                                                     \
+            def_##moduleName.m_methods = &this->_modules[moduleID][0];                                                 \
+            if (PyImport_AppendInittab(getExtensionModuleName(moduleID), &PyInit_##moduleName) == -1) {                \
+                setError("PyImport_AppendInittab failed to append " #moduleName);                                      \
+                return 1;                                                                                              \
+            }                                                                                                          \
+        }                                                                                                              \
+    }
+
 class Three : public Six {
 public:
+    // Python module constants need to be added in the init callback after calling
+    // "PyModule_Create". The constants need to be globally available.
+    static PyModuleConstants ModuleConstants;
+    static std::mutex ModuleConstantsMtx;
```

Comment:
In case we add multiple constant from different thread, it might be overkill but I don't think the performance really matters since it's only use at initialization.

Reply:
The library is not threadsafe, if we go down this path there's a lot that need to be put behind a mutex, current design assumes the caller will take care of that